Documentation/v2.md

```diff
@@ -0,0 +1,48 @@
+# Dex v2
+
+## Streamlined deployments
+
+Many of the changes between v1 and v2 were aimed at making dex easier to deploy and manage, perhaps the biggest pain point for dex v1. Dex is now a single, scalable binary with a sole source of configuration. Many components which previously had to be set through the API, such as OAuth2 clients and IDP connectors can now be specified statically. The new architecture lacks a singleton component eliminating deployment ordering. There are no more special development modes; instructions for running dex on a workstation translate with minimal changes to a production system.
+
+All of this results in a much simpler deployment story. Write a config file, run the dex binary, and that's it.
+
+## More storage backends
+
+Dex's internal storage interface has been cleaned up and simplified, allowing dex to use more backing databases to store its state. A notable addition uses the Kubernetes API (a key value store) to store state, allowing dex to run natively on top of Kubernetes clusters.
```

Comment:
I would reword this section to be as follows:

Dex's internal storage interface has been improved to support multiple backing databases including **Postgres**, **SQLite3**, and the **Kubernetes API**.  The Kubernetes API, a `[key value store][link-to-k8s-api-docs]`, allows dex to be run natively on top of a Kubernetes cluster.  Starting with support for multiple storage backends should ensure that the Dex storage interface is *actually* pluggable.

A more in depth discussion on existing storage options and how to add new ones can be found in the `[storage documentation][storage-docs]`.